app/uk/gov/hmrc/timetopayproxy/connectors/TtpConnector.scala

```diff
@@ -95,34 +87,22 @@ class DefaultTtpConnector @Inject()(appConfig: AppConfig, httpClient: HttpClient
                   implicit ec: ExecutionContext,
                   hc: HeaderCarrier
                 ): TtppEnvelope[UpdatePlanResponse] = {
-    val path = if(appConfig.useIf) "individuals/time-to-pay/quote" else "debts/time-to-pay/quote"
+    val path = if (appConfig.useIf) "individuals/time-to-pay/quote" else "debts/time-to-pay/quote"
     val url = s"${appConfig.ttpBaseUrl}/$path/${updatePlanRequest.customerReference.value}/${updatePlanRequest.planId.value}"
 
-    val response: Future[Either[ConnectorError, UpdatePlanResponse]] =
+    EitherT(
       httpClient
```

Comment:
TtppError is afaik the Error that the proxy respond. However the TimeToPayError is the Error that Time To Pay returns. since this is making a call to Time to Pay, the error should not be TtppError but rather TimeToPayError

Reply:
Happy to resolve this based on our conversation?

Reply:
It was my confusion, I confused TtppError with TimeToPayErrorResponse. However TtppError is the trait of the internal error model representation. Ignore my comment please. Your implementation is the way it should be. Resolved.